templates/gmetad.conf.erb

```diff
@@ -102,12 +102,16 @@ gridname "<%= scope.lookupvar('::ganglia::gmetad::gridname') %>"
 # is always trusted.
 # default: There is no default value
 # trusted_hosts 127.0.0.1 169.229.50.165 my.gmetad.org
+trusted_hosts <%= @trusted_hosts.join(' ') %>
```

Comment:
Joshua Hoblitt <notifications@github.com> writes:

>> @@ -102,12 +102,16 @@ gridname "<%= scope.lookupvar('::ganglia::gmetad::gridname') %>"
>>  # is always trusted.
>>  # default: There is no default value
>>  # trusted_hosts 127.0.0.1 169.229.50.165 my.gmetad.org
>> +trusted_hosts <%= @trusted_hosts.join(' ') %>

> This causes failures under the future parser (which are allowed to fail
> in the travis matrix due to a change in erb `:undef` handing) -- could
> you add a condition check for nil and check the output of the future
> parser runs?

        I've added the condition check for nil?, along with some other
checks.  I'm running into an issue with the .spec bit for a block like
this, though:

   context 'default' do
     it 'should have an empty trusted_hosts' do
       should contain_file('/etc/ganglia/gmetad.conf').
         with_content(/^# trusted_hosts .*$/)
     end
   end # default

        It's complaining like the 'trusted_hosts' block can't be parsed,
but I'm really not sure why.  I've left it commented out for now; the
checks are still *fairly* useful but not as good as I'd like.

        As for the future parser, it's still griping, but apparently based
on something to do with case_sensitive_hostnames.

        I'm happy to take another round at this if you think I can help.

                           - Tim Skirvin (tskirvin@killfile.org)
-- 
http://wiki.killfile.org/           Skirv's Homepage <FISH>< <*>
http://www.flickr.com/photos/tskirvin           Skirv's Pictures